aspnetcore/grpc/comparison.md

```diff
@@ -30,28 +30,28 @@ This article provides a comparison between gRPC and HTTP APIs, and recommends sc
 
 ### Performance
 
-gRPC messages are serialized using [Protobuf](https://developers.google.com/protocol-buffers/docs/overview), an efficient binary message format. Protobuf is very fast to serialize on the server and client, and it results in small message payloads, important in limited bandwidth scenarios like mobile applications.
+gRPC messages are serialized using [Protobuf](https://developers.google.com/protocol-buffers/docs/overview), an efficient binary message format. Protobuf serializes very quickly on the server and client. Protobuf serialization results in small message payloads, important in limited bandwidth scenarios like mobile apps.
 
-Addtionally gRPC is designed for HTTP/2, a major revision of HTTP that provides significant performance benefits over HTTP 1.x:
+gRPC is designed for HTTP/2, a major revision of HTTP that provides significant performance benefits over HTTP 1.x:
 
-* Binary framing and compression. HTTP/2 protocol is compact and efficent to send and recieve.
+* Binary framing and compression. HTTP/2 protocol is compact and efficient sending and receiving.
 * Multiplexing of multiple HTTP/2 calls over a single TCP connection. Multiplexing eliminates [head-of-line blocking](https://en.wikipedia.org/wiki/Head-of-line_blocking).
 
 ### Code Generation
 
-All gRPC frameworks provide first-class support for code generation. A core file to gRPC development is the [`*.proto` file](https://developers.google.com/protocol-buffers/docs/proto3), which defines the contact of gRPC services and messages. From this file gRPC frameworks will code generate a service base class, messages and a complete client.
+All gRPC frameworks provide first-class support for code generation. A core file to gRPC development is the [`*.proto` file](https://developers.google.com/protocol-buffers/docs/proto3), which defines the contact of gRPC services and messages. From this file gRPC frameworks will code generate a service base class, messages, and a complete client.
 
-By sharing the `*.proto` file between the server and client you can generate messages and client code from end to end. This eliminates duplication when building clients and can be a significant time saver.
+By sharing the `*.proto` file between the server and client, messages and client code can be generated from end to end. This sharing eliminates duplication when building clients and can be a significant time saver.
```

Comment:
The sharing of proto files alone isn't what that saves time, it is the code generation of the client.

How about:

```suggestion
By sharing the `*.proto` file between the server and client, messages and client code can be generated from end to end. Code generation of the client eliminates duplication of messages on the client and server, and creates a strongly-typed client for you. Not having to write a client saves significant development time in applications with many services.
```